File: src/KnightFrank.Antares.Dal/Model/Property/Viewing.cs
namespace KnightFrank.Antares.Dal.Model.Property
{
    using System;
    using System.Collections.Generic;

    using KnightFrank.Antares.Dal.Model.Contacts;
    using Activities;
    using KnightFrank.Antares.Dal.Model.User;

    public class Viewing : BaseEntity
    {
        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public string InvitationText { get; set; }

        public string PostViewingComment { get; set; }

        public Guid ActivityId { get; set; }

        public virtual Activity Activity { get; set; }

        public Guid RequirementId { get; set; }

        public virtual Requirement Requirement { get; set; }

        public Guid NegotiatorId { get; set; }

        public virtual User Negotiator { get; set; }

        public List<Contact> Attendees { get; set; } = new List<Contact>();
    }
}

File: src/wwwroot/app/common/models/business/viewing/viewing.ts
/// <reference path="../../../../typings/_all.d.ts" />

module Antares.Common.Models.Business {
    declare var moment: any;

    export class Viewing implements Dto.IViewing {
        id: string = null;
        activityId: string = null;
        requirementId: string = null;
        negotiatorId: string = null;
        negotiator: User;
        startDate: Date | string = null;
        endDate: Date | string = null;
        invitationText: string;
        postViewingComment: string;
        attendeesIds: string[];
        attendees: Contact[];
        day: string;
        activity: Business.Activity;
        requirement: Business.Requirement;

        constructor(viewing?: Dto.IViewing) {
            if (viewing) {
                angular.extend(this, viewing);

                if (viewing.attendees) {
                    this.attendees = viewing.attendees.map((contact: Dto.IContact) => new Contact(contact));
                }

                if (viewing.attendees) {
                    this.negotiator = new User(viewing.negotiator);
                }

                this.day = Core.DateTimeUtils.getDatePart(viewing.startDate);
                if (viewing.requirement) {
                    this.requirement = new Requirement(viewing.requirement);
                }
                if (viewing.activity) {
                    this.activity = new Business.Activity(viewing.activity);
                }
                this.negotiator = new User(viewing.negotiator);
                this.startDate = moment(viewing.startDate).toDate();
                this.endDate = moment(viewing.endDate).toDate();
            }
        }
    }
}